juju/juju
# 2.4 enable ha no presence check

## Description of change

This changes `juju enable-ha` to no longer demote/remove controller machines.
This simplifies the enable-ha logic significantly, and means we don't need some of our old workarounds (like forcing the machine we are running on to not be treated as unavailable).

## QA steps

```
$ juju bootstrap
$ juju enable-ha
$ kill/stop machine 1
$ juju enable-ha
```
Should no longer try to remove machine 1 from being a controller, and won't start a machine 3. Instead, doing:
```
$ juju remove-machine 1
$ juju enable-ha
```
Should give the user the ability to do exactly as they requested.

## Documentation changes

This is one of the major changes of 2.4 vs previous releases. We're helping users to be explicit in how they want to change the controllers that are running, instead of hoping the side-effect of `juju enable-ha` would do what they want.

## Bug reference

[lp:1658033](https://bugs.launchpad.net/juju/+bug/1658033)
[lp:1703040](https://bugs.launchpad.net/juju/+bug/1703040)
